gradle/docker.gradle

```diff
@@ -0,0 +1,66 @@
+buildscript {
+    repositories {
+        jcenter()
+
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
+    }
+
+    dependencies {
+        classpath 'com.bmuschko:gradle-docker-plugin:3.1.0'
+        classpath "com.avast.gradle:gradle-docker-compose-plugin:0.6.1"
+    }
+}
+
+apply plugin: com.avast.gradle.dockercompose.DockerComposePlugin
+
+dockerCompose.isRequiredBy(test)
+
+dockerCompose {
+    useComposeFiles = ['docker/docker-compose.yml']
+}
+
+test.doFirst {
+
+
+    // exposes "${serviceName}_HOST" and "${serviceName}_TCP_${exposedPort}" environment variables
+    // for example exposes "WEB_HOST" and "WEB_TCP_80" environment variables for service named `web` with exposed port `80`
+    // if service is scaled using scale option, environment variables will be exposed for each service instance like "WEB_1_HOST", "WEB_1_TCP_80", "WEB_2_HOST", "WEB_2_TCP_80" and so on
+    dockerCompose.exposeAsEnvironment(test)
+    // exposes "${serviceName}.host" and "${serviceName}.tcp.${exposedPort}" system properties
+    // for example exposes "web.host" and "web.tcp.80" system properties for service named `web` with exposed port `80`
+    // if service is scaled using scale option, environment variables will be exposed for each service instance like "web_1.host", "web_1.tcp.80", "web_2.host", "web_2.tcp.80" and so on
+    dockerCompose.exposeAsSystemProperties(test)
+    // get information about container of service `web` (declared in docker-compose.yml)
```

Comment:
are these comments relevant?

Reply:
It depends on CircleCI – if it's ok for us to run the docker-compose with well known ports (80, 8080, all the ones S3 uses), there's no need. Otherwise we will have to use the things described here to let docker assign random ports and use this to set those ports as properties to the test process.